apps/wallet/src/routes/Deposit/Buy.tsx

```diff
@@ -1,23 +1,38 @@
-import { useEffect, useRef, useState } from 'react';
-import { ErrorPage } from '../../ErrorPage';
-import { TRANSAK_API_KEY, TRANSAK_URL, account } from '../../config';
-import { Transak } from './transak/constants';
+import { colors } from '@telegram-wallet-ui/twa-ui-kit';
 import { MainButton } from '@twa-dev/sdk/react';
+import { useEffect, useRef, useState } from 'react';
 import { useNavigate, useParams } from 'react-router-dom';
+import { ErrorPage } from '../../ErrorPage';
 import { Page } from '../../components';
+import { TRANSAK_API_KEY, TRANSAK_URL, account } from '../../config';
+import { getDebug } from '../../lib/utils/debug';
 import { URLParams } from '../../types';
+import { Transak } from './transak/constants';
+
+const debug = getDebug('Buy');
 
 const walletAddress = account?.address;
 
-const constructSrcUrl = (walletAddress: string, tokenSymbol: string) => {
+const convertToTransakCurrencyCode = (tokenSymbol: string) => {
```

Comment:
If I understand correctly, this would mean that the user selected wmatic in the app, and will buy MATIC through transak?

If that's the case, what we wanted is to filter out WMATIC from the coin list when the user is depositing via bank card.

This is dependant on "Deposit flow - choose between bank card/external before choosing token" that @sukhsinghcodes is handling, otherwise you won't be able to tell.

Reply:
This will be resolved in a subsequent PR. The user will first have to choose credit card deposit, then will choose from a subset of coins that are supported by Transax - WBTC,WETH,USDT,USDC. MATIC will not be shown.